api/src/main/java/io/minio/SourceInfo.java

```diff
@@ -0,0 +1,99 @@
+package io.minio;
+
+import io.minio.errors.InvalidArgumentException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/** SourceInfo creates the SourceObject
+ * used for Compose Object.
+ */
+public class SourceInfo {
+  private String bucketName;
+  private String objectName;
+  private int start;
+  private int end;
+  private ServerSideEncryption sseSource;
+  private Map<String, String> headerMap = new HashMap<>();
+
+    /**
+     * Constucts the SourceInfo with bucket name and object name.
+     */
+  public SourceInfo(String bucketName, String objectName ) {
+    this.bucketName = bucketName;
+    this.objectName = objectName;
+    this.start = -1;
+    headerMap.put("x-amz-copy-source", S3Escaper.encodePath(bucketName + "/"
+            + objectName));
+  }
+    /**
+     * Constucts the SourceInfo with bucket name, object name and Server Side Encryption .
+     */
+  public SourceInfo(String bucketName, String objectName, ServerSideEncryption sseSource) {
+    this.bucketName = bucketName;
+    this.objectName = objectName;
+    this.sseSource = sseSource;
+    this.start = -1;
+    headerMap.put("x-amz-copy-source", S3Escaper.encodePath(bucketName + "/"
+            + objectName));
+  }
+
+  public String getBucketName() {
+    return bucketName;
+  }
+
+  public void setBucketName(String bucketName) {
```

Comment:
You could have below constructors
```
CopySource(String bucketName, String objectName)
CopySource(String bucketName, String objectName, int offset, int length)
CopySource(String bucketName, String objectName, Integer offset, Integer length, Map<String, String> headerMap)
CopySource(String bucketName, String objectName, Integer offset, Integer length, ServerSideEncryption sse, Map<String, String> headerMap)
```
and no setters are required.  here `headerMap` carries all headers below where user sets
```
x-amz-copy-source-range:bytes=first-last
x-amz-copy-source-if-match: etag
x-amz-copy-source-if-none-match: etag
x-amz-copy-source-if-unmodified-since: time_stamp
x-amz-copy-source-if-modified-since: time_stamp
```
This way there is no need to have condition methods.